### PR TITLE
fixes some bugs with product and redirecting on signup

### PR DESCRIPTION
**Pull request by arunbhardwaj**
also makes itemcard accept a children header component

### CODE REVIEW

File: components/ItemCard/FavoriteHeader.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const handleFavoritedItem = () => {
+    setFavorited(() => !isFavorited);
+  };
+
+  return (
+    <CardHeader
+      sx={{ zIndex: 9999 }}
+      action={
+        <IconButton
+          aria-label="add to favorites"
```
Comment (kellytso04) [inline components/ItemCard/FavoriteHeader.js:15]: consider changing this to `aria-label='add-to-favorites'`

File: components/ItemCard/index.js [hunk 2]
```diff
@@ -8,42 +8,19 @@ import CardImage from "./CardImage.js";
...
-      {product?.createdBy?.email}
-      <UsernameButton />
-      {/* {product?.createdBy && (
-        <Link href={`/users/${product?.createdBy?._id}`}>
-          {product?.createdBy?.email}
-        </Link>
-      )} */}
-    </Card>
+    <Link href={`/products/${product._id}`}>
+      <Card raised sx={{ width: "15vw", cursor: "pointer" }}>
```
Comment (kellytso04) [inline components/ItemCard/index.js:15]: i think we should wrap the `Card` component in `CardActionArea` for semantic purposes

Comment (kellytso04) [reply components/ItemCard/index.js:15]: in fact i strongly believe we should!

File: components/ItemCard/index.js [hunk 3]
```diff
@@ -8,42 +8,19 @@ import CardImage from "./CardImage.js";
...
+        {children}
+        <CardImage />
+        <CardDescription />
+        <Link href={`/products/${product._id}`}>link</Link>
```
Comment (kellytso04) [inline components/ItemCard/index.js:19]: did you intend to keep _link_ here?

File: screens/ProductList/index.js [hunk 4]
```diff
@@ -3,15 +3,22 @@ import ItemCard from "@components/ItemCard";
...
 import { useSearchContext } from "@context/productSearch";
+import FavoriteHeader from "@components/ItemCard/FavoriteHeader";
 
 const ProductPage = () => {
   const [products, setProducts] = useState([]);
   const { searchedProducts, setSearchedProducts } = useSearchContext();
 
   const fetchProducts = async () => {
     try {
+<<<<<<< HEAD
```
Comment (kellytso04) [inline screens/ProductList/index.js:14]: leftover file conflict stuff here

File: src/helper.js [hunk 5]
```diff
@@ -8,6 +8,8 @@ export const register = (email, password) => {
   createUserWithEmailAndPassword(auth, email, password)
     .then((userCredential) => {
       const user = userCredential.user;
+      // console.log(userCredential)
```
Comment (kellytso04) [inline src/helper.js:11]: don't forget to delete these after debug

**Review by kellytso04 — APPROVED** [review]

File: components/ItemCard/index.js (hunk 2 above)
Comment (arunbhardwaj) [reply components/ItemCard/index.js:15]: DONE ITS SEXY

File: components/ItemCard/index.js (hunk 3 above)
Comment (arunbhardwaj) [reply components/ItemCard/index.js:19]: nope, this was michael's code, I already commented it out

File: components/ItemCard/FavoriteHeader.js (hunk 1 above)
Comment (arunbhardwaj) [reply components/ItemCard/FavoriteHeader.js:15]: ✔️

**Pull request merged**